minifi/minifi-toolkit/minifi-toolkit-configuration/src/main/java/org/apache/nifi/minifi/toolkit/configuration/json/TransformCommandFactory.java

```diff
@@ -0,0 +1,117 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.minifi.toolkit.configuration.json;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import org.apache.nifi.controller.flow.VersionedDataflow;
+import org.apache.nifi.minifi.toolkit.configuration.ConfigMain;
+import org.apache.nifi.minifi.toolkit.configuration.ConfigTransformException;
+import org.apache.nifi.minifi.toolkit.configuration.PathInputStreamFactory;
+import org.apache.nifi.minifi.toolkit.configuration.PathOutputStreamFactory;
+import org.apache.nifi.registry.flow.RegisteredFlowSnapshot;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.module.jakarta.xmlbind.JakartaXmlBindAnnotationIntrospector;
+
+public class TransformCommandFactory {
+
+    public static final String TRANSFORM = "transform";
+
+    private static final String COMMAND_DESCRIPTION = "Transform NiFi flow JSON format into MiNifi flow JSON format";
+    private static final String PROPERTY_KEY_VALUE_DELIMITER = "=";
+
+    private final PathInputStreamFactory pathInputStreamFactory;
+    private final PathOutputStreamFactory pathOutputStreamFactory;
+
+    public TransformCommandFactory(PathInputStreamFactory pathInputStreamFactory,
+            PathOutputStreamFactory pathOutputStreamFactory) {
+        this.pathInputStreamFactory = pathInputStreamFactory;
+        this.pathOutputStreamFactory = pathOutputStreamFactory;
+    }
+
+    public ConfigMain.Command create() {
+        return new ConfigMain.Command(this::transformToJson, COMMAND_DESCRIPTION);
+    }
+
+    private int transformToJson(String[] args) {
+        if (args.length != 3) {
+            printTransformUsage();
+            return ConfigMain.ERR_INVALID_ARGS;
+        }
+
+        String sourceNiFiJsonPath = args[1];
+        String targetMiNiFiJsonPath = args[2];
+
+        try {
+            RegisteredFlowSnapshot registeredFlowSnapshot = readNifiFlow(sourceNiFiJsonPath);
+            VersionedDataflow versionedDataflow = new VersionedDataflow();
+            versionedDataflow.setRootGroup(registeredFlowSnapshot.getFlowContents());
+            persistFlowJson(versionedDataflow, targetMiNiFiJsonPath);
+        } catch (ConfigTransformException e) {
```

Comment:
We should also copy the parameter contexts from the exported flow.
Without this if an exported flow contains parameter referenes, then the transformed flow will fail as it won't find the related parameter context.
```
versionedDataflow.setParameterContexts(new ArrayList<>(registeredFlowSnapshot.getParameterContexts().values()));
```